test/suite/test_checkpoint33.py

```diff
@@ -0,0 +1,108 @@
+# OTHER DEALINGS IN THE SOFTWARE.
+
+from test_cc01 import test_cc_base
+from suite_subprocess import suite_subprocess
+import time
+from wiredtiger import stat
+from wtdataset import SimpleDataSet
+
+# test_checkpoint33.py
+#
+class test_checkpoint33(test_cc_base, suite_subprocess):
+    create_params = 'key_format=i,value_format=S,allocation_size=4KB,leaf_page_max=32KB,'
+    conn_config = 'cache_size=100MB,statistics=(all),statistics_log=(wait=1,json=true,on_close=true)'
+    uri = 'table:test_checkpoint33'
+
+    table_numkv = 10000
+    value_size = 1024
+    value = 'a' * value_size
+
+    def delete(self):
+        c = self.session.open_cursor(self.uri, None)
+        for k in range(self.table_numkv):
+            c.set_key(k)
+            self.session.begin_transaction()
+            c.remove()
+            self.session.commit_transaction("commit_timestamp=" + self.timestamp_str(4))
+        c.close()
+
+    def populate(self):
+        c = self.session.open_cursor(self.uri, None)
+        for k in range(self.table_numkv):
+            self.session.begin_transaction()
+            c[k] = self.value
+            self.session.commit_transaction("commit_timestamp=" + self.timestamp_str(2))
+        c.close()
+
+    def get_size(self):
+        c = self.session.open_cursor('statistics:' + self.uri, None, None)
+        file_size = c[stat.dsrc.block_size][2]
+        c.close()
+        return(file_size)
+
+    def test_checkpoint33(self):
+        # Pin oldest timestamp 1.
+        self.conn.set_timestamp(f'oldest_timestamp={self.timestamp_str(1)}')
+
+        # Create and populate a table at timestamp 2.
+        self.session.create(self.uri, self.create_params)
+        self.session.checkpoint()
+        self.populate()
+
+        # Make everything stable at timestamp 3.
+        self.conn.set_timestamp(f'stable_timestamp={self.timestamp_str(3)}')
+
+        # Write to disk.
+        self.session.checkpoint()
+
+        # Delete everything at timestamp 4.
+        self.delete()
+
+        # Make the deletions stable at timestamp 5.
+        self.conn.set_timestamp(f'stable_timestamp={self.timestamp_str(5)}')
+
+        # Write to disk.
+        self.session.checkpoint()
+
+        # Make everything globally visible.
+        self.conn.set_timestamp(f'oldest_timestamp={self.timestamp_str(5)}')
+        self.prout(f'File size: {self.get_size()}')
+
+        # Give checkpoint cleanup enough opportunity to clean up all the deleted pages.
+        cc_success = 0
+        while (cc_success < 10):
+            self.session.checkpoint('debug=(checkpoint_cleanup=true)')
+            c = self.session.open_cursor( 'statistics:')
+            cc_success = c[stat.conn.checkpoint_cleanup_success][2]
+            c.close()
+            self.prout(f'cc_success={cc_success}')
+            self.prout(f'File size: {self.get_size()}')
+            time.sleep(0.1)
+
+        self.assertLessEqual(self.get_size(), 12 * 1024)
```

Comment:
If it's equal, does it mean the file size has not been reduced at all?

Reply:
Sorry I actually meant to call this out. For the current paramters this is the minimum size the file will reduce to. The file will be the size of the block manager checkpoint blocks. If the number of k/v's in the test changes this seems to change with it and I'm not sure why.